lib/config/hadeployconfigverify.go

```diff
@@ -0,0 +1,271 @@
+package config
+
+import (
+	"container/list"
+	"errors"
+)
+
+func (c *HaDeployConfig) Verify(configFile string) error {
+	errorList := list.New()
+
+	// Check if config file path is empty
+	if len(configFile) < 1 {
+		return errors.New("config file path is empty")
+	}
+
+	// Parse the config file
+	haDeployConfig, err := c.Parse(configFile)
+	if err != nil {
+		errorList.PushBack(err.Error())
+	} else {
+		c = haDeployConfig
+	}
+
+	if haDeployConfig == nil || haDeployConfig.Architecture == nil {
+		return errors.New("invalid ha deploy config")
+	}
+
+	// Validate ExistingInfra (on prem)
+	if haDeployConfig.Architecture.ExistingInfra != nil {
+		c.verifyConfigInitials(haDeployConfig.Architecture.ExistingInfra, errorList)
+		c.validateExistingInfraBackupConfig(errorList)
+		c.verifyExistingInfraSettings(haDeployConfig.ExistingInfra.Config, errorList)
+
+		// on prem aws or self-managed
+		if isExternalDb(haDeployConfig) {
+			c.verifyExternalPgSettings(haDeployConfig.External.Database.PostgreSQL, errorList)
+			c.verifyExternalOsSettings(haDeployConfig.External.Database.OpenSearch, errorList)
+			c.verifyAwsExternalOsSettings(haDeployConfig.External.Database.OpenSearch.Aws, errorList)
```

Comment:
This is only valid for cases where external AWS OpenSearch DB is configured. In case of non-AWS deployment, these check is going to fail unnecessarily. We need to add condition to run check only on that special case.